src/pip/_internal/cli/cmdoptions.py

```diff
@@ -244,6 +244,33 @@ class PipOption(Option):
     help="Disable prompting for input.",
 )
 
+keyring_provider: Callable[..., Option] = partial(
+    Option,
+    "--keyring-provider",
+    dest="keyring_provider",
+    choices=["disabled", "import", "subprocess"],
+    default="disabled",
```

Comment:
I think we should add an `auto` choice to do the try-import-and-then-subprocess fallback logic, and keep that as the default for compatibility (either permanently, or for a transition period).